nncf/quantization/algorithms/channel_alignment/backend.py

```diff
@@ -0,0 +1,169 @@
+        :return: Bias value in the NumPy format.
+        """
+
+    @staticmethod
+    @abstractmethod
+    def get_activation_port_ids_for_node(node: NNCFNode) -> Tuple[int, int]:
+        """
+        Returns Input Port ID and Output Port ID corresponding to activation input and output edges for
+        the node.
+        Supports only nodes that could have bias value.
+
+        :param node: Node of NNCFGraph with bias value.
+        """
+
+    @staticmethod
+    @abstractmethod
+    def get_weights_port_ids_for_node(node: NNCFNode) -> Tuple[int, int]:
+        """
+        Returns Input Port ID and Output Port ID corresponding to node weights inputr port id and
+        constant output port id the node.
+
+        :param node: Node of NNCFGraph.
+        """
+
+    @staticmethod
+    @abstractmethod
+    def get_statistic_collector(
+        reduction_shape, q: float, num_samples: int, inplace: bool
+    ) -> TensorStatisticCollectorBase:
+        """
+        Get backend-specific tensor collector that collects medians of minimal and maximal quantiles.
+
+        :param reduction_shape: Target reduction shape for the reduction.
+        :param q: Minimal quantile for the tensor collector.
+        :param num_samples: Num samples to collect by the tensor collector.
+        :param inplace: Should statistic be calculated inplace or out of place.
+        :return: Backend-specific tensor collector that collects medians of minimal and maximal quantiles.
+        """
+
+    @staticmethod
+    @abstractmethod
+    def is_node_with_bias(node: NNCFNode, nncf_graph: NNCFGraph) -> bool:
+        """
+        Checks if the node has a bias or not.
+
+        :param node: The node to check.
+        :param nncf_graph: The NNCF graph.
+        :return: True` if `node` corresponds to the operation with bias
+            (bias is added to the output tensor of that operation), `False` otherwise.
+        """
+
+    @staticmethod
+    @abstractmethod
+    def create_bias_update_command(node_with_bias: NNCFNode, updated_value: np.ndarray, nncf_graph: NNCFGraph):
+        """
+        Creates backend-specific command to update bias value.
+
+        :param node: The node for which bias should be updated.
+        :param bias_value: New value for the bias.
+        :param nncf_graph: NNCFGraph of the target model.
+        :return: Backend-specific command to update bias value.
+        """
+
+    @staticmethod
+    @abstractmethod
+    def create_weights_update_command(node_with_weights: NNCFNode, updated_value: np.array, weights_port_id: int):
+        """
+        Creates backend-specific command to update convolution or linear layer
+        weights.
+
+        :param node: The node for which weights should be updated.
+        :param updated_value: New weights for the node.
+        :param weights_port_id: Target node weights input port id.
+        :return: Backend-specific command to update bias value.
+        """
```

Comment:
Could you please use `CommandCreator` abstraction to reduce number of function in the backend class?
https://github.com/openvinotoolkit/nncf/blob/a1bec7147a675a6bf76b5a386c8660e36d4aba71/nncf/common/graph/transformations/command_creation.py#L21

You can create instance of the `CommandCreator` class inside common code using

https://github.com/openvinotoolkit/nncf/blob/a1bec7147a675a6bf76b5a386c8660e36d4aba71/nncf/common/factory.py#L99

Reply:
We should consider to do the same for BC algos

Reply:
Done, please check